pep-9999.rst

```diff
@@ -0,0 +1,588 @@
+PEP: 9999
+Title: Support for parsing TOML in the Standard Library
+Author: Taneli Hukkinen, Shantanu Jain <hauntsaninja at gmail.com>
+Sponsor: TODO
+PEP-Delegate: TODO
+Discussions-To: TODO
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 01-Jan-2022
+Python-Version: 3.11
+Post-History: 1900-01-01
+
+
+Abstract
+========
+
+This proposes adding a module, ``tomllib``, to the standard library for
+parsing TOML. [1]_
+
+
+Motivation
+==========
+
+The TOML format is the format of choice for Python packaging, as evidenced by
+:pep:`517`, :pep:`518` and :pep:`621`. Including TOML support in the standard
+library helps avoid bootstrapping problems for Python build tools. Currently
+most Python build tools need to vendor a TOML parsing library.
+
+Python tools are increasingly configurable via TOML, for examples: ``black``,
+``mypy``, ``pytest``, ``tox``, ``pylint``, ``isort``. Those that are not, such
+as ``flake8``, cite the lack of standard library support as a `main reason why
+<https://github.com/PyCQA/flake8/issues/234#issuecomment-812800657>`_.
+
+Given the special place TOML already has in the Python ecosystem, it makes sense
+for this to be an included battery.
+
+Finally, TOML as a format is increasingly popular (some reasons for this are
+outlined in PEP 518). Hence this is likely to be a generally useful addition,
+even looking beyond the needs of Python packaging and Python tooling: various
+Python TOML libraries have about 2000 reverse dependencies on PyPI. For
+comparison, ``requests`` has about 28k reverse dependencies.
+
+
+Rationale
+=========
+
+This PEP proposes basing the standard library support for reading TOML on the
+third party library ``tomli`` [2]_.
+
+Many projects have recently switched to using ``tomli``, for example, ``pip``,
+``build``, ``pytest``, ``mypy``, ``black``, ``flit``, ``coverage``,
+``setuptools-scm``, ``cibuildwheel``.
+
+``tomli`` is actively maintained and well-tested. ``tomli`` is about 800
+lines of code with 100% test coverage and passes all tests in the official TOML
+compliance test suite.
```

Comment:
Note that the official TOML compliance test suite is very much a work in progress: there are 0 test cases currently merged.

There is a PR https://github.com/toml-lang/compliance/pull/8 to add test cases from https://github.com/iarna/toml-spec-tests. These are the tests that Tomli passes.

Tomli also passes tests in https://github.com/BurntSushi/toml-test. That test set has been updated quite recently though so I'm not sure if that is true with the latest revision.

To be more accurate, we should maybe refer to this PR https://github.com/toml-lang/compliance/pull/8 here. I'll also try to find the time to make sure Tomli passes the most recent BurntSushi tests and then maybe we mention that here as well.